files/entrypoint.sh

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env bash
+
+
+# Variables
+# --
+#
+# These values can be overridden by ENV Vars
+#
+K8S_VERSION=${K8S_VERSION:-v1.6.1}
+KUBEADM_RELEASE=${KUBEADM_RELEASE:-v1.6.0-alpha.0.2074+a092d8e0f95f52}
+CNI_RELEASE=${CNI_RELEASE:-07a8a28637e97b22eb8dfe710eeae1344f69d16e}
+
+ARCH=${ARCH:-amd64}
+CNI_BIN_DIR=${CNI_BIN_DIR:-/opt/cni}
+ROOTFS=${ROOTFS:-/rootfs}
+
+
+# Kubernetes Google Cloud Storage
+# --
+#
+# Handling usage of Dev / Final release buckets
+#
+K8S_URL=${K8S_URL:-https://storage.googleapis.com/kubernetes-release/release}
+CNI_URL=${CNI_URL:-https://storage.googleapis.com/kubernetes-release/network-plugins}
+
+echo "Installing kubeadm version: ${KUBEADM_RELEASE}"
+if [[ -z "${KUBEADM_URL}" ]]; then
+  if [[ -n "${KUBEADM_VERSION/v[0-9].[0-9].[0-9]/}" ]]; then
+    echo "Kubeadm: is a Git version"
+    KUBEADM_URL=https://storage.googleapis.com/kubernetes-release-dev/ci-cross
+  else
+    echo "Kubeadm: is a Release version"
+    KUBEADM_URL=https://storage.googleapis.com/kubernetes-release/release
+  fi
+fi
+echo "Kubeadm: Download url - ${KUBEADM_URL}"
+
+
+# Linux Distro
+# --
+#
+# Environment settings according to Distro
+#
+if [[ $1 == "coreos" ]]; then
```

Comment:
I thing we can drop the distinguishment between CoreOS release channels if we manage also the deployment of the `kubelet-wrapper`: 

https://coreos.com/kubernetes/docs/latest/kubelet-wrapper.html